src/init.cpp

```diff
@@ -1573,6 +1562,18 @@ bool AppInit2(Config &config, thread_group &threadGroup)
     if (GetBoolArg("-listenonion", DEFAULT_LISTEN_ONION))
         StartTorControl(threadGroup);
 
+    // Startup txindex just before StartNode. If we start it earlier and before ActivateBestChain
+    // we can end up grinding slowly through ActivateBestChain when txindex still has unfinished
+    // compaction to do from a prior run.
+    fTxIndex = GetBoolArg("-txindex", DEFAULT_TXINDEX);
+    if (fTxIndex)
+    {
+        uiInterface.InitMessage(_("Starting txindex"));
+        auto txindex_db = new TxIndexDB(cacheConfig.nTxIndexCache, false, GetBoolArg("-reindex", DEFAULT_REINDEX));
```

Comment:
why not use fReindex as the 3rd arg?

Reply:
I believe this was correct but I added some clarifying code and comments.

> why not use fReindex as the 3rd arg?

I thinks this is correct but I added some clarifying comments and code.